src/EditorFeatures/TestUtilities/Threading/WpfTestCase.cs

```diff
@@ -19,6 +20,13 @@ public class WpfTestCase : XunitTestCase
     {
         private readonly SemaphoreSlim _wpfTestSerializationGate;
 
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        [Obsolete("Called by the de-serializer; should only be called by deriving classes for de-serialization purposes")]
+        public WpfTestCase()
+        {
+            _wpfTestSerializationGate = new SemaphoreSlim(1);
```

Comment:
General comment is I don't understand the architecture here to say if this is safe, but my gut is saying this should be kept to null, so that way RunTest() crashes should somebody every try to call this on a deserialized object. If there's a reason that it *does* need to be callable on a deserialized object, then the architecture here is broken (possibly by our own doing) and we need to revisit.